processor/src/chiplets/aux_trace/mod.rs

```diff
@@ -242,23 +170,111 @@ where
+
+        match op_code {
+            JOIN | SPLIT | LOOP | DYN | CALL => {
+                build_control_block_request(main_trace, op_code_felt, alphas, row)
+            }
+            SYSCALL => build_syscall_block_request(main_trace, op_code_felt, alphas, row),
+            SPAN => build_span_block_request(main_trace, alphas, row),
+            RESPAN => build_respan_block_request(main_trace, alphas, row),
+            END => build_end_block_request(main_trace, alphas, row),
+            AND => build_bitwise_request(main_trace, ZERO, alphas, row),
+            XOR => build_bitwise_request(main_trace, ONE, alphas, row),
+            MLOADW => build_mem_request_word(main_trace, MEMORY_READ_LABEL, alphas, row),
+            MSTOREW => build_mem_request_word(main_trace, MEMORY_WRITE_LABEL, alphas, row),
+            MLOAD => build_mem_request_element(main_trace, MEMORY_READ_LABEL, alphas, row),
+            MSTORE => build_mem_request_element(main_trace, MEMORY_WRITE_LABEL, alphas, row),
+            MSTREAM => build_mstream_request(main_trace, alphas, row),
+            RCOMBBASE => build_rcomb_base_request(main_trace, alphas, row),
+            HPERM => build_hperm_request(main_trace, alphas, row),
+            MPVERIFY => build_mpverify_request(main_trace, alphas, row),
+            MRUPDATE => build_mrupdate_request(main_trace, alphas, row),
+            _ => E::ONE,
+        }
+    }
+
+    /// Constructs the responses from the chiplets to the other VM-components at `row`.
+    fn get_responses_at(&self, main_trace: &MainTrace, alphas: &[E], row: usize) -> E
+    where
+        E: FieldElement<BaseField = Felt>,
+    {
+        let selector0 = main_trace.chiplet_selector_0(row);
+        let selector1 = main_trace.chiplet_selector_1(row);
+        let selector2 = main_trace.chiplet_selector_2(row);
+        let selector3 = main_trace.chiplet_selector_3(row);
+        let selector4 = main_trace.chiplet_selector_4(row);
+
+        if selector0 == ZERO {
+            build_hasher_chiplet_responses(main_trace, row, alphas, selector1, selector2, selector3)
+        } else if selector1 == ZERO {
+            debug_assert_eq!(selector0, ONE);
+            build_bitwise_chiplet_responses(main_trace, row, selector2, alphas)
+        } else if selector2 == ZERO {
+            debug_assert_eq!(selector0, ONE);
+            debug_assert_eq!(selector1, ONE);
+            build_memory_chiplet_responses(main_trace, row, selector3, alphas)
+        } else if selector3 == ZERO {
+            debug_assert_eq!(selector0, ONE);
+            debug_assert_eq!(selector1, ONE);
+            debug_assert_eq!(selector2, ONE);
+            build_kernel_chiplet_responses(main_trace, row, selector4, alphas)
+        } else {
+            debug_assert_eq!(selector0, ONE);
+            debug_assert_eq!(selector1, ONE);
+            debug_assert_eq!(selector2, ONE);
+            debug_assert_eq!(selector3, ONE);
+            E::ONE
+        }
+    }
```

Comment:
Similarly, I think this would be clearer if we had different functions: `is_hasher_chiplet(s0, s1, s2, s3)`, `is_memory_chiplet(s0, s1, s2, s3)`, etc.